src/main/java/de/medizininformatikinitiative/torch/util/ReferenceHandler.java

```diff
@@ -166,61 +159,94 @@ public Flux<List<ResourceGroup>> handleReference(ReferenceWrapper referenceWrapp
                         .flatMap(List::stream)
                         .toList()
                 )
-                .flatMapMany(list -> {
-                    ResourceAttribute referenceAttribute = referenceWrapper.toResourceAttributeGroup();
-                    if (referenceWrapper.refAttribute().mustHave() && list.isEmpty()) {
-                        processingBundle.setResourceAttributeInValid(referenceAttribute);
-                        return Flux.error(new MustHaveViolatedException(
-                                "MustHave condition violated: No valid references were resolved for " + referenceWrapper.references()
-                        ));
+                .flatMapMany(list -> checkReferenceViolatesMustHave(referenceWrapper, list, processingBundle));
+    }
+
+    /**
+     * Collects all valid resourceGroups
+     *
+     * <p> For a given reference and resource checks if already a valid group in processingBundle.
```

Comment:
please write a complete sentence. ... if there is already ... ... in the {@code processingBundle}.

Reply:
No I ment: 

> For a given reference and resource checks if already a valid group in processingBundle.

Isn't a sentence.